Documentation.md

```diff
@@ -4,20 +4,91 @@
+| Initial Line   | (^(?=.*[a-zA-Z].*)([a-zA-Z0-9]*)$)? | Checks if last_value is true                               | Start Conditional? |
+| Internal Logic |                                     | If the conditional is true, this line(s) will be executed. |                    |
+| Final Line     | (^(?=.*[a-zA-Z].*)([a-zA-Z0-9]*)$). | Concludes the conditional statement                        | End Conditional.   |
+
+## For Loops (Foreach loops)
+
+For loops in Pyoem are similar to conditionals in the sense of inital and final statements, however there is an additional line for dictating what is being looped over.
+
+| Section Type   | Definition                            | Description                                                  | Example        |
+| -------------- | ------------------------------------- | ------------------------------------------------------------ | -------------- |
+| Initial Line   | (^(?=.*[a-zA-Z].*)([a-zA-Z0-9]*)$)... | Starts the loop section                                      | Start Loop...  |
+| Loop Logic     | i, value, list                        | Defines the variables i and value, and provides the list to loop over | i, value, list |
+| Internal Logic |                                       | If the loop logic is true, this line(s) will be executed.    |                |
+| Final Line     | (^(?=.*[a-zA-Z].*)([a-zA-Z0-9]*)$).   | Ends the loop section                                        | End Loop.      |
+
+## Function Calls
+
+Pyoem supports the ability for developers to write their own functions and call them in other stanzas. Function declarations must be written before the main stanza that calls it. The name of the function corresponds to the values used from the parameters provided, if you have 3 parameters, the function name must have three words. You don't have to use all the words in the function name as parameters but your function name must have at least two words in it. The structure of a function is as follows: 
+
+| Section Type   | Definition      | Description                                                  | Example       |
+| -------------- | --------------- | ------------------------------------------------------------ | ------------- |
+| Title          | ^(.*?[A-Z]){2,} | The title is made of at least 2 word that are all capitalized | FUNCTION CALL |
+| Internal Logic |                 | If the function is called, this line(s) will be executed.    |               |
+
+When calling a function the structure follows "parameter1, paramter2" (Function Call). The (Function Call) section references the function you made previously. The (Function Call) doesn't need to be capitalized here. The "parameter1, parameter2" indicates the list of parameters, for this example in the Function Call function, parameter1 maps to function and parameter 2 maps to call. 
```

Comment:
Found typo, "paramter2" should be changed to "parameter2"

Reply:
Thanks!